### PR TITLE
[Hub Generated] Review request for Microsoft.Consumption to add version stable/2019-11-01

### PR DESCRIPTION
**Pull request by sofisa16**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sofisa16/azure-rest-api-specs/sopadill-newAPI-consumption-Microsoft.Consumption-2019-11-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sofisa16** [comment]
I am fixing these errors:

https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=28726C76-E136-4F6B-896F-20733648963D
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=9672F2DA-2A14-4F38-B3C4-D0E8F3437911
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=0519C5A4-04E8-4EC9-8E45-75F59792570D
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=BA1D830F-B0B4-4A69-AE23-F8A2A6D2A69B
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=9F7EE149-644B-4753-961F-E4DCCCA7B49C
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=750F0740-506E-4C71-9C75-3FC2F9FEEDE5
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=20BA3AEE-9D7B-4F8B-ABDB-E7F9B8815706

**Comment by jorgecotillo** [comment]
While adding new version, the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch/PR with this recommendation for faster review.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    "Tag": {
+      "description": "The tag resource.",
+      "properties": {
+        "key": {
+          "description": "Tag key.",
+          "type": "string"
+        }
+      }
+    },
+    "ErrorDetails": {
```
Comment (jorgecotillo) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:146]: Consider using types under common-types

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+          "description": "Resource type."
+        },
+        "eTag": {
+          "type": "string",
+          "description": "eTag of the resource. To handle concurrent update scenario, this field will be used to determine whether the user is updating the latest version or not."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "Operation": {
```
Comment (jorgecotillo) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:196]: I don't see this being used nor as part of your example, is this needed?

Comment (sofisa16) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:201]: Yes, it is done.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      }
+    }
+  },
+  "parameters": {
+    "scopeTagsParameter": {
+      "name": "scope",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The scope associated with tags operations. This includes '/subscriptions/{subscriptionId}/' for subscription scope, '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}' for resourceGroup scope, '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}' for Billing Account scope, '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/departments/{departmentId}' for Department scope, '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/enrollmentAccounts/{enrollmentAccountId}' for EnrollmentAccount scope and '/providers/Microsoft.Management/managementGroups/{managementGroupId}' for Management Group scope..",
```
Comment (filizt) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:242]: >..", [](start = 676, length = 4)

nit - double period at the end

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            "eTag": {
+              "type": "string",
+              "description": "eTag of the resource. To handle the concurrent update scenario, this field will be used to determine whether the user is updating the latest version or not."
+            }
+          }
+        }
+      ],
+      "x-ms-azure-resource": true
+    },
+    "Operation": {
```
Comment (filizt) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:186]: Operation model seems to be missing some required properties (such as isDataAction, description). Please see more here: [https://github.com/Azure/azure-resource-manager-rpc/blob/eec9f32c0357d0421be13c82ced767943013901a/v1.0/proxy-api-reference.md#exposing-available-operations](https://github.com/Azure/azure-resource-manager-rpc/blob/eec9f32c0357d0421be13c82ced767943013901a/v1.0/proxy-api-reference.md#exposing-available-operations)

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    "Tag": {
+      "description": "The tag resource.",
+      "properties": {
+        "key": {
+          "description": "Tag key.",
+          "type": "string"
+        }
+      }
+    },
+    "ErrorDetail": {
```
Comment (filizt) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:151]: >ErrorDetail [](start = 5, length = 11)

I can't see any references for ErrorDetail definition. Please remove if not being used.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```
Comment (majastrz) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:36]: >paths [](start = 3, length = 5)

The diff between the 2019-10-01 (first commit) and last commit is fairly large. It appears that 2019-11-01 will not include a lot of the funcitonality. Can you explain?

**Comment by sofisa16** [comment]
I am fixing all the errors of this PR and addressing the first comment left: 

> While adding new version, the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch/PR with this recommendation for faster review.

in this new PR
https://github.com/Azure/azure-rest-api-specs/pull/12400

**Pull request closed without merging**